lib/stdlib/src/gen_server.erl

```diff
@@ -19,87 +19,95 @@
 
--                     -----> Module:terminate/2
+    -                     -----> Module:terminate/2
 
--                     -----> Module:code_change/3
-```
+    -                     -----> Module:code_change/3
+    ```
 
-If a callback function fails or returns a bad value, the `gen_server` process
-terminates.
+    If a callback function fails or returns a bad value,
+    the `gen_server` process terminates.  However, an exception of class
+    [`throw`](`erlang:throw/1`) is not regarded as an error
+    but as a valid return, from all callback functions.
 
-A `gen_server` process handles system messages as described in `m:sys`. The
-`m:sys` module can be used for debugging a `gen_server` process.
+    A `gen_server` process handles system messages as described in `m:sys`.
+    The `m:sys` module can be used for debugging a `gen_server` process.
 
-Notice that a `gen_server` process does not trap exit signals automatically,
-this must be explicitly initiated in the callback module.
+    Notice that a `gen_server` process does not trap exit signals
+    automatically, this must be explicitly initiated in the callback module.
 
-Unless otherwise stated, all functions in this module fail if the specified
-`gen_server` process does not exist or if bad arguments are specified.
+    Unless otherwise stated, all functions in this module fail
+    if the specified `gen_server` process does not exist
+    or if bad arguments are specified.
 
-The `gen_server` process can go into hibernation (see `erlang:hibernate/3`) if a
-callback function specifies `'hibernate'` instead of a time-out value. This can
-be useful if the server is expected to be idle for a long time. However, use
-this feature with care, as hibernation implies at least two garbage collections
-(when hibernating and shortly after waking up) and is not something you want to
-do between each call to a busy server.
+    The `gen_server` process can go into hibernation (see `erlang:hibernate/3`)
+    if a callback function specifies `'hibernate'` instead of a time-out value.
+    This can be useful if the server is expected to be idle for a long time.
+    However, use this feature with care, as hibernation implies at least
+    two garbage collections (when hibernating and shortly after waking up)
+    and is not something you want to do between each call to a busy server.
 
-If the `gen_server` process needs to perform an action immediately after
-initialization or to break the execution of a callback into multiple steps, it
-can return `{continue,Continue}` in place of the time-out or hibernation value,
-which will immediately invoke the [`handle_continue/2`](`c:handle_continue/2`)
-callback.
+    If the `gen_server` process needs to perform an action immediately after
+    initialization or to break the execution of a callback into multiple steps,
+    it can return `{continue,Continue}` in place of
+    the time-out or hibernation value, which will immediately invoke
```

Comment:
> I didn't look too closely, but it looks like that technically the handle_continue call will be done when the gen_server re-enters the main loop. In that regard, it would be more accurate to say that handle_continue is called before any other (message-caused) events are handled.

I guess that is what is meant by "immediately"; before receiving a message.

But as you say, other ordered or implied actions are performed first.  I'll rephrase according to your suggestion.

Reply:
> There are also several double spaces.

I follow an ancient e-mail convention of having double spaces after a full stop.  To be able do distinguish from e.g. an abbreviation...

I hope that doesn't break some newer guidelines; Markdown nevertheless ignores such double spaces so they don't get through to the resulting HTML.